src/pages/patientView/clinicalInformation/PatientViewPageStore.ts

```diff
@@ -69,39 +69,68 @@ export class PatientViewPageStore
 
     @observable.ref private visibleRows:Mutation[][] = [];
 
-    @observable patientId = '';
+    @observable private _patientId = '';
+    @computed get patientId():string {
+        if (this._patientId)
+            return this._patientId;
+
+        return this.derivedPatientId.result;
+    }
 
     @observable studyId = '';
 
-    @observable sampleId = 'P04_Pri';
+    @observable _sampleId = '';
+    @computed get sampleId() {
+        return this._sampleId;
+    }
 
-    @observable pageMode: PageMode = 'patient';
+    @computed get pageMode(): PageMode {
+        return this._sampleId ? 'sample' : 'patient';

```

Comment:
There is a small possibility that _sampleId may be null when set from outside, and I prefer not crashing in that case (in other parts of the code)